packages/common/src/ui/components/navigation/NavigationPrompt/NavigationPrompt.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect } from 'react';
+// import { Prompt } from 'react-router-dom';
+
+export interface NavigationPromptProps {
+  isUnsaved: boolean;
+}
+
+export const NavigationPrompt: React.FC<NavigationPromptProps> = ({
```

Comment:
Is this a component because `react-router-dom` might eventually provide a component and so it would make it easy to change later?

I wondered if it would be a little easier if this was a hook

```
const { isDirty, ..., ..., ..., } = useDraftOutboundLines();
useAreYouSureYouWantToLeave(isDirty)
```

something like that? 🤷

Reply:
it is - if there is a situation where there isn't a blocking modal, like we currently have, then the navigation with `react-router` won't trigger the browser prompt. I don't know when that might happen, or if we'll actually need it. I'm thinking now that I'll use a hook and comment to warn people instead